From: jiyun_ha <jiyun_ha@tmax.co.kr>
Date: Tue, 1 Jun 2021 11:28:11 +0900
Subject: [feat] tos-hotkey : make widget close after session changed

 - when the system turned into lock-dialog session the widget does remain, so added session mode signal handler to deal with such situation

[IMS] N/A
---
 extensions/tos-hotkey/baseWidget.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extensions/tos-hotkey/baseWidget.js b/extensions/tos-hotkey/baseWidget.js
index 79fb463..5dddffe 100644
--- a/extensions/tos-hotkey/baseWidget.js
+++ b/extensions/tos-hotkey/baseWidget.js
@@ -163,6 +163,8 @@ var BaseWidget = GObject.registerClass({ // eslint-disable-line no-unused-vars
 
         this._grabHelper = new GrabHelper.GrabHelper(this);
         Main.layoutManager.connect('system-modal-opened', () => this.close());
+
+        Main.sessionMode.connect('updated', () => this.close());
     }
 
     get state() {
